### PR TITLE
[3.2] What's new

### PR DESCRIPTION
**Pull request by MaryaBelanger**
PRs from August 17 to present (will publish after 3.2 release so link to new pages can be added)

### CODE REVIEW

**Comment by parlough** [comment]
Telling me "Do not review" makes me want to review it even more!!

**Review by atsansone — APPROVED** [review]
LGTM

File: src/guides/whats-new.md [hunk 1]
```diff
@@ -17,6 +17,63 @@ and follow the [Dart blog][].
 [dart-announce]: https://groups.google.com/a/dartlang.org/d/forum/announce
 [Dart blog]: https://medium.com/dartlang
 
+## November 15, 2023: 3.2 release
+
+This section lists notable changes made from August 17, 2023,
+through November 15, 2023.
+For details about the 3.2 release,
+check out [3.2 blog post][] and the and the [SDK changelog][3-2-changelog].
```
Comment (parlough) [inline src/guides/whats-new.md:25]:
```suggestion
check out [3.2 blog post][] and the [SDK changelog][3-2-changelog].
```

File: src/guides/whats-new.md [hunk 2]
```diff
@@ -17,6 +17,63 @@ and follow the [Dart blog][].
...
+
+[3.2 blog post]: /
```
Comment (parlough) [inline src/guides/whats-new.md:27]:
```suggestion
[3.2 blog post]: https://medium.com/dartlang/dart-3-2-c8de8fe1b91f
```

File: src/guides/whats-new.md [hunk 3]
```diff
@@ -17,6 +17,63 @@ and follow the [Dart blog][].
...
+[3-2-changelog]: https://github.com/dart-lang/sdk/blob/main/CHANGELOG.md#320
+
+### Docs updated or added to dart.dev
+{: .no_toc}
+
+In addition to bug fixes and incremental improvements,
+we made the following changes to this site:
+
+* Updated the [Fixing type promotion failures][no-promo] page
+  for a new type promotion feature, private final field promotion.
```
Comment (parlough) [inline src/guides/whats-new.md:37]:
```suggestion
  for a new type promotion feature: private final field promotion.
```

File: src/guides/whats-new.md [hunk 4]
```diff
@@ -17,6 +17,63 @@ and follow the [Dart blog][].
...
+* Accounted for new and updated [lints][] and [diagnostics][] in their
+  respective doc pages.
+* Added documentation for the new `--tighten ` flag to the [`pub upgrade`] page.
+* Removed the cheatsheet in favor of the [Language overview][] page.
+* Clarified the relationship between [guard clauses and patterns][].
+* Adjusted the [Constructors][] page to better represent best practices.
+* Improved contents of the [Package dependencies][] page to be more actionable
+  and easier to follow.
+* Elaborated on static members in the [Extension methods][] page.
+* Changed the [Objective C][] multithreading limitations content to account
```
Comment (parlough) [inline src/guides/whats-new.md:53]:
```suggestion
* Changed the [Objective-C][] multithreading limitations content to account
```

File: src/guides/whats-new.md [hunk 5]
```diff
@@ -17,6 +17,63 @@ and follow the [Dart blog][].
...
+[Breaking changes]: /resources/breaking-changes
+[lints]: /tools/linter-rules
+[diagnostics]: /tools/diagnostic-messages
+[`pub upgrade`]: /tools/pub/cmd/pub-upgrade#--tighten
+[Language overview]: /language
+[guard clauses and patterns]: /language/patterns#switch-statements-and-expressions
+[Constructors]: /language/constructors
+[Package dependencies]: /tools/pub/dependencies
+[Extension methods]: /language/extension-methods
+[Objective C]: /guides/libraries/objective-c-interop#callbacks-and-multithreading-limitations
```
Comment (parlough) [inline src/guides/whats-new.md:73]:
```suggestion
[Objective-C]: /guides/libraries/objective-c-interop#callbacks-and-multithreading-limitations
```

File: src/guides/whats-new.md [hunk 6]
```diff
@@ -17,6 +17,63 @@ and follow the [Dart blog][].
...
+
+In addition to bug fixes and incremental improvements,
+we made the following changes to this site:
+
+* Updated the [Fixing type promotion failures][no-promo] page
+  for a new type promotion feature, private final field promotion.
+  * Made minor related adjustments to the Effective Dart entry on
+    [type promotion][], the [Understanding Null Safety][] page, and other various
+    places across the site.
+* Introduced documentation on the experimental native assets feature
```
Comment (parlough) [inline src/guides/whats-new.md:41]: Consider linking directly to https://dart.dev/guides/libraries/c-interop#native-assets

**Review by parlough — APPROVED** [review]
Thanks @MaryaBelanger! I'm happy to now know what's new.

A few minor suggestions:

**Pull request merged**